client/components/data/document-head/index.jsx

```diff
@@ -65,6 +76,10 @@ class DocumentHead extends Component {
 				this.props.addMeta( meta );
 			} );
 		}
+
+		if ( nextProps.formattedTitle !== this.props.formattedTitle ) {
```

Comment:
[The docs](https://facebook.github.io/react/docs/component-specs.html#updating-componentwillreceiveprops) really aren't too clear on this. [This](https://github.com/facebook/react/blob/5514ea369d7f60127ae15774e1276b21cd7c0107/src/renderers/dom/server/__tests__/ReactServerRendering-test.js#L126-L186) is the best confirmation I could find.